src/core/mempool.rs

```diff
@@ -956,11 +1114,124 @@ impl MemPoolDB {
         }
     }
 
+    /// Remove all txids at the given height from the bloom counter.
+    /// Used to clear out txids that are now outside the bloom counter's depth.
+    fn prune_bloom_counter(tx: &mut MemPoolTx, target_height: u64) -> Result<(), MemPoolRejection> {
+        let sql = "SELECT txid FROM mempool WHERE height = ?1 AND NOT EXISTS (SELECT 1 FROM removed_txids WHERE txid = mempool.txid)";
```

Comment:
Is this query implementable as a `JOIN`, e.g.,

```
SELECT A.txid FROM mempool as A LEFT OUTER JOIN removed_txids as B on A.txid = B.txid WHERE B.txid IS NULL AND A.height = ?1
```

which I think is usually more efficient than inner selects. Of course, this may be totally unnecessary optimization.

Reply:
(Same comment applies to other cases of the inner select)